pymc/distributions/timeseries.py

```diff
@@ -33,6 +37,93 @@
 ]
 
 
+class GaussianRandomWalkRV(RandomVariable):
+    """
+    GaussianRandomWalk Random Variable
+    """
+
+    name = "GaussianRandomWalk"
+    ndim_supp = 1
+    ndims_params = [0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("GaussianRandomWalk", "\\operatorname{GaussianRandomWalk}")
+
+    def _shape_from_params(self, dist_params, reop_param_idx=1, param_shapes=None):
+        # (size which is number of time series, steps)
+        return (dist_params[-2], dist_params[-1])
+        raise Exception("Ravin's shape exception")
+
+        # if self.ndim_supp <= 0:
+        #     raise ValueError("ndim_supp must be greater than 0")
+        # if param_shapes is not None:
+        #     ref_param = param_shapes[rep_param_idx]
+        #     return (ref_param[-self.ndim_supp],)
+        # else:
+        #     ref_param = dist_params[rep_param_idx]
+        #     if ref_param.ndim < self.ndim_supp:
+        #         raise ValueError(
+        #             (
+        #                 "Reference parameter does not match the "
+        #                 f"expected dimensions; {ref_param} has less than {self.ndim_supp} dim(s)."
+        #             )
+        #         )
+        #     return ref_param.shape[-self.ndim_supp:]
+
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        mu: Union[np.ndarray, float],
+        sigma: Union[np.ndarray, float],
+        init: float,
+        steps: int,
+        size: int,
+    ) -> np.ndarray:
+        """Gaussian Random Walk generator.
+
+        The init value is drawn from the Normal distribution with the same sigma as the
+        innovations.
+
+        Notes
+        -----
+        Currently does not support custom init distribution
+
+        Parameters
+        ----------
+        rng: np.random.RandomState
+           Numpy random number generator
+        mu: np.ndarray
+           Random walk mean
+        sigma: np.ndarray
+            Standard deviation of innovation (sigma > 0)
+        init: float
+            Initialization value for GaussianRandomWalk
+        steps: int
+            Length of random walk, must be greater than 1. Returned array will be of size+1 to
+            account as first value is initial value
+        size: int
+            The number of Random Walk time series generated
+
+        Returns
+        -------
+        np.ndarray
+        """
+
+        if steps is None or steps == 0:
+            raise ValueError("Steps must be greater than 0 or not None")
+        if size is None:
+            size = 1
+
+        init_val = rng.normal(init, sigma, size=(size, 1))
+        steps = rng.normal(loc=mu, scale=sigma, size=(size, steps))
```

Comment:
This is all a bit confusing to me, but would this make more sense?
```suggestion
        steps = rng.normal(loc=mu, scale=sigma, size=(size, steps-1))
```

Reply:
I see you resolved this.  Does this makes sense now?